test_files/KPOINTS_SrBa2Sn2O7_bands

```diff
@@ -0,0 +1,16 @@
+K-points for bandstructure G-X-P-N-G-M
```

Comment:
If possible, it would be great to avoid adding large new test files. Would it be possible to reuse an existing `vasprun.xml` in your test? Maybe `lifepo4.xml`?

Reply:
Is the lifepo4.xml generated with LORBIT>=11, and is there a corresponding KPOINTS file? Or I can upload another structure with a smaller .xml file. Does that work? What will be the maximum size you can accept?

Reply:
No, `lifepo4.xml ` has `LORBIT` `F`. Since you're testing plotting code and not file parsing, maybe you could serialize the `pymatgen` via their `as_dict` methods and save the output as JSON, then reload it with `from_fict` in the test?

Reply:
Looks numpy related. Maybe todo with the 'fixes' we applied after the `numpy` v.1.24.0 breaking changes. This test case should trigger it though:

https://github.com/materialsproject/pymatgen/blob/44f24db50d3b59ad9bf448c523564c114f472f19/pymatgen/electronic_structure/tests/test_bandstructure.py#L215-L221

What `numpy` version are you running?

I just checked out your PR and ran the updated test. No error on my end.

```py

        band_struct = v.get_band_structure(
            kpoints_filename=os.path.join(PymatgenTest.TEST_FILES_DIR, "KPOINTS_SrBa2Sn2O7_bands")
        )
        print(f"{sorted(band_struct.as_dict())=}")
```

prints

```
sorted(band_struct.as_dict())=['@class', '@module', 'band_gap', 'bands', 'branches', 'cbm', 'efermi', 'is_metal', 'is_spin_polarized', 'kpoints', 'labels_dict', 'lattice_rec', 'projections', 'structure', 'vbm']
```

Reply:
I'm using v.1.24.0.
The `as_dict` function works fine for me. The error comes out when I dump the dict to a json file.

Reply:
Oh yes, my bad. Sorry it's a bit late here. You could pass a custom `handler` to `json.dump` to convert to normal `int`. Something like (untested)

```py
def handler(obj: Any):
    if isinstance(obj, np.integer):
        return int(obj)
```

Though I'm wondering if maybe that should be part of `BS.as_dict`. Writing to JSON is a common reason far `as_dict`.

Reply:
Hi Janosh, I replaced the .xml file with the smaller .json file.